cloudflare/resource_cloudflare_custom_hostnames.go

```diff
@@ -0,0 +1,223 @@
+			},
+			"ssl": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"method": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"http", "email", "cname"}, false),
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"dv"}, false),
+						},
+						"custom_hostname_settings": {
+							Type:     schema.TypeList,
+							MaxItems: 4,
+							Optional: true,
+							ForceNew: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"http2": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"on", "off"}, false),
+									},
+									"tls13": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"on", "off"}, false),
+									},
+									"mintlsversion": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"1.0", "1.1", "1.2", "1.3"}, false),
+									},
+									"ciphers": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceCloudflareCustomHostnamesCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	log.Printf("[DEBUG] zone ID: %s", zoneID)
+	zcso := expandToZoneCustomHostnamesOptions(d)
+	res, err := client.CreateCustomHostname(zoneID, zcso)
+	if err != nil {
+		return fmt.Errorf("Failed to create custom hostname cert: %s", err)
+	}
+
+	if res.Result.ID == "" {
+		return fmt.Errorf("Failed to find custom ssl in Create response: id was empty")
+	}
+
+	d.SetId(res.Result.ID)
+
+	log.Printf("[INFO] Cloudflare Custom SSL ID: %s", d.Id())
+
+	return resourceCloudflareCustomHostnamesRead(d, meta)
+}
+
+func resourceCloudflareCustomHostnamesUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	certID := d.Id()
+	zcso := expandToZoneCustomHostnamesOptionsSSL(d)
+
+	log.Printf("[INFO] Updating Custom Hostname from struct: %+v", zcso)
+
+	_, err := client.UpdateCustomHostnameSSL(zoneID, certID, zcso)
+	if err != nil {
+		return errors.Wrap(err, "Error updating Custom Hostname")
+	}
+
+	return resourceCloudflareCustomHostnamesRead(d, meta)
+}
+
+func resourceCloudflareCustomHostnamesRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Why aren't the `ssl` option changes reflected in the `Read`? If the remote changes them for any reason, it's not reflected here.